egress/src/reports/windrose.rs

```diff
@@ -0,0 +1,1114 @@
+                ARRAY_AGG((speed.obs, direction.obs)::windobs) \
+            FROM ( \
+                SELECT obstime, corrected AS obs FROM legacy.data \
+                WHERE timeseries = $1 \
+                AND corrected IS NOT NULL \
+                AND corrected > -30000.0 \
+                AND quality_code IS NOT NULL \
+                AND quality_code != 7 \
+                AND EXTRACT(minute FROM obstime)::int = 0 \
+                AND obstime >= $4 AND obstime < $5 \
+                AND ($3::int[] IS NULL OR EXTRACT(month FROM obstime)::int = ANY($3)) \
+            ) speed \
+            INNER JOIN ( \
+                SELECT obstime, corrected AS obs FROM legacy.data \
+                WHERE timeseries = $2 \
+                AND corrected IS NOT NULL \
+                AND corrected > -30000.0 \
+                AND quality_code IS NOT NULL \
+                AND quality_code != 7 \
+                AND EXTRACT(minute FROM obstime)::int = 0 \
+                AND obstime >= $4 AND obstime < $5 \
+                AND ($3::int[] IS NULL OR EXTRACT(month FROM obstime)::int = ANY($3)) \
+            ) direction \
+            USING (obstime) \
+            GROUP BY day",
+        )
+        .await?;
+
+    let mut futures = patches
+        .iter()
+        .map(|patch| async {
+            conn.query(
+                &query,
+                &[
+                    &patch.speed_tsid,
+                    &patch.direction_tsid,
+                    &months,
+                    &patch.from,
+                    &patch.to,
+                ],
+            )
+            .await
+        })
+        .collect::<FuturesOrdered<_>>();
+
+    let mut days = Vec::new();
+    while let Some(res) = futures.next().await {
+        let rows = res?;
+
+        for row in rows {
+            days.push(WindDay {
+                observations: row.get(1),
+            });
+        }
+    }
+
+    Ok(days)
+}
+
+#[derive(Clone, Default, PartialEq, Debug)]
+struct WindPatch {
+    speed_tsid: i64,
+    direction_tsid: i64,
+    from: DateTime<Utc>,
+    to: DateTime<Utc>,
+}
+
+/// Merge the speed and direction timeseries patches
+// TODO: is there a better algorithm?
+// Both vectors should be quite small, so probably this is good enough
+fn merge_patches(speeds: Vec<Patch>, directions: Vec<Patch>) -> Vec<WindPatch> {
+    if speeds.is_empty() || directions.is_empty() {
+        return vec![];
+    }
+
+    let patches = speeds
+        .iter()
+        .flat_map(|speed| {
+            directions
+                .iter()
+                // Only keep patches that overlap
+                .filter(|direction| speed.from < direction.to && speed.to > direction.from)
+                .map(|direction| WindPatch {
+                    speed_tsid: speed.tsid,
+                    direction_tsid: direction.tsid,
+                    from: speed.from.max(direction.from),
+                    to: speed.to.min(direction.to),
+                })
```

Comment:
I'm not sure this implementation is correct? Anyway it's perhaps better to apply the `Timerange.overlap` function from the patchwork module in a `filter_map`?

Reply:
On a second read it looks alright, but I still think it's not ideal to replicate this logic. I noticed `Timerange` takes `Option`s, perhaps we need types for open/closed timeranges, and use the closed impl in the open impl?

Reply:
Implemented overlap for `Patch`